pkgs/development/python-modules/cirq/default.nix

```diff
@@ -0,0 +1,87 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchFromGitHub
+, google_api_python_client
+, matplotlib
+, networkx
+, numpy
+, pandas
+, protobuf  # pkgs.protobuf, not pythonPackages.protobuf
+, pythonProtobuf  # pythonPackages.protobuf
+, requests
+, scipy
+, sortedcontainers
+, sympy
+, typing-extensions
+  # test inputs
+, pytestCheckHook
+, pytest-benchmark
+, ply
+, pydot
+, pyyaml
+, pygraphviz
+}:
+
+buildPythonPackage rec {
+  pname = "cirq";
+  version = "0.6.1";
+
+  disabled = pythonOlder "3.5";
+
+  # TODO: replace fetchFromGitHub with fetchPypi in future if don't need patching
+  # Pypi doesn't host source code for Cirq, only wheels, which doesn't work with requirements patch
+  src = fetchFromGitHub {
+    owner = "quantumlib";
+    repo = "cirq";
+    rev = "v${version}";
+    sha256 = "0lhr2dka7vpz9xd6akxphrcv2b3ni2cgjywpc1r7qpqa5mrq1q7f";
+  };
+
+  # Cirq 0.6 requires networkx==2.3 only for qiskit dependency/test, force this to avoid networkx version conflicts
```

Comment:
Is there some upstream issue we could reference here?

Reply:
https://github.com/quantumlib/Cirq/issues/2368. Will include in comment